### PR TITLE
[2.2] Backporting #4414: Themes should use `theme.yml` now, fallback to old `config.yml`

### PR DESCRIPTION
**Pull request by bobdenotter**

### CODE REVIEW

File: src/Config.php [hunk 1]
```diff
@@ -364,7 +364,12 @@ protected function parseContentTypes(array $generalConfig)
      */
     protected function parseTheme($themePath, array $generalConfig)
     {
-        $themeConfig = $this->parseConfigYaml('config.yml', $themePath);
+        $themeConfig = $this->parseConfigYaml('theme.yml', $themePath);
+
+        /** @deprecated Since 2.3 and will be removed in Bolt v3.0 (config.yml was the old filename) */
```
Comment (GawainLynch) [inline src/Config.php]: So we can remove this in master :wink:

Comment (bobdenotter) [inline src/Config.php]: Ah, no.. I used the checkout from _before_ the version bump to do the backport.. Will fixe these numbers.

File: tests/codeception/acceptance/102-BackendDeveloper/BackendDeveloperCest.php [hunk 2]
```diff
@@ -99,7 +99,7 @@ public function fileManagementViewEditTemplatesTest(\AcceptanceTester $I)
         $I->see('css/',         Locator::href("/bolt/files/theme/$dir/css"));
         $I->see('images/',      Locator::href("/bolt/files/theme/$dir/images"));
         $I->see('javascripts/', Locator::href("/bolt/files/theme/$dir/javascripts"));
-        $I->see('config.yml',   Locator::href("/bolt/file/edit/theme/$dir/config.yml"));
+        $I->see('theme.yml',    Locator::href("/bolt/file/edit/theme/$dir/theme.yml"));
```
Comment (bobdenotter) [inline tests/codeception/acceptance/102-BackendDeveloper/BackendDeveloperCest.php]: Go home, Nitpick-CI!

Comment (GawainLynch) [inline tests/codeception/acceptance/102-BackendDeveloper/BackendDeveloperCest.php]: "…you're drunk!"

Comment (alsanchez86) [inline tests/codeception/acceptance/102-BackendDeveloper/BackendDeveloperCest.php]: Really?

Comment (GawainLynch) [inline tests/codeception/acceptance/102-BackendDeveloper/BackendDeveloperCest.php]: Ha! Not @bobdenotter but @Nitpick-CI has had a few :beers: (just joking, on all points)

**Comment by GawainLynch** [comment]
You'd almost thing the :koala: wrote the Travis test file…

**Pull request merged**